### PR TITLE
Simavr 1.3

### PR DESCRIPTION
**Pull request by goodrone**
Include simavr from https://github.com/buserror/simavr

Include self as a maintainer.

Tested on NixOS.

### CODE REVIEW

**Comment by joachifm** [comment]
But you already have https://github.com/NixOS/nixpkgs/pull/15681

**Comment by goodrone** [comment]
@joachifm yes, I was planning to drop #15681.  Is this a bad idea?

**Comment by joachifm** [comment]
@goodrone seems easier to just update the existing PR; that also has the benefit of retaining context, minimizing the need for clicking around to find related discussion.

File: pkgs/development/tools/simavr/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  libPath = stdenv.lib.makeLibraryPath [
+    libelf
+  ];
+
+  src = fetchurl {
+    url = "https://github.com/buserror/simavr/archive/v${version}.tar.gz";
+    sha1 = "b1a875fb23ad819545bcb1889e5fb5470deb09c5";
+  };
+
+  unpackCmd = ''
```
Comment (groxxda) [inline pkgs/development/tools/simavr/default.nix]: Why is this needed?

File: pkgs/development/tools/simavr/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  version = "1.3";
+  enableParallelBuilding = true;
+
+  libPath = stdenv.lib.makeLibraryPath [
+    libelf
+  ];
+
+  src = fetchurl {
+    url = "https://github.com/buserror/simavr/archive/v${version}.tar.gz";
+    sha1 = "b1a875fb23ad819545bcb1889e5fb5470deb09c5";
```
Comment (groxxda) [inline pkgs/development/tools/simavr/default.nix]: Is there a need for sha1 (weak)? If possible please update to sha256. 
Also `fetchFromGitHub` might be the perfect match for this job ;-)

**Comment by bjornfor** [comment]
Duplicate: https://github.com/NixOS/nixpkgs/pull/15681

**Comment by goodrone** [comment]
@joachifm I didn't find a way to change my PR to use my topic branch instead of `master` (was my mistake originally).  Please let me know if I should close this and return to the original PR, but this one is better for me to work on.  Thank you!

File: pkgs/development/tools/simavr/default.nix (hunk 1 above)
Comment (goodrone) [inline pkgs/development/tools/simavr/default.nix]: @groxxda thanks.  I feel like I've chosen a not very suitable package as a template for my package.

File: pkgs/development/tools/simavr/default.nix (hunk 2 above)
Comment (goodrone) [inline pkgs/development/tools/simavr/default.nix]: @groxxda thank you!

**Comment by joachifm** [comment]
@goodrone aha, then it makes sense.

File: pkgs/development/tools/simavr/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    repo = "simavr";
+    rev = "51d5fa69f9bc3d62941827faec02f8fbc7e187ab";
+    sha256 = "0k8xwzw9i6xsmf98q43fxhphq0wvflvmzqma1n4jd1ym9wi48lfx";
+  };
+
+  buildFlags = "AVR_ROOT=${avrgcclibc}/avr SIMAVR_VERSION=${version}";
+  installFlags = buildFlags + " DESTDIR=$(out)";
+
+  postFixup = ''
+    patchelf --set-rpath "$libPath:$out/lib" "$out/bin/simavr"
```
Comment (joachifm) [inline pkgs/development/tools/simavr/default.nix]: It is odd that simply linking against `libelf` doesn't make it show up in the rpath. Do you know why that is? Also, won't this remove every other lib from the rpath?

Comment (goodrone) [inline pkgs/development/tools/simavr/default.nix]: @joachifm thank you.  Now I don't blindly overwrite the rpath, but add `$out/lib` to it.  And got rid of explicit libelf dependency.

**Comment by goodrone** [comment]
Travis CI build failed - `error: preallocating file of 20568773 bytes: No space left on device`

**Comment by siddharthist** [comment]
@goodrone @joachifm What's the status of this PR? I don't see anything specific holding it up.

**Comment by joachifm** [comment]
The man-db patch is unrelated and should be its own PR, but other than that I guess this can be integrated.

**Comment by siddharthist** [comment]
@joachifm Would you mind cherry-picking?

**Comment by joachifm** [comment]
@siddharthist sure, I've added it to my merge queue.

**Comment by joachifm** [comment]
@goodrone I've integrated simavr for Linux, but left out the darwin stuff. Please re-submit those changes against master if you still wish to have them integrated. Thanks.

**Comment by bjornfor** [comment]
Did master change drastically since this PR was created? I'm getting build failure:

```
$ nix-build -A simavr  # on master (0e69fb2f196d553e1db04aff46f50868aec42424)
...
LD obj-x86_64-unknown-linux-gnu/charlcd.elf
charlcd done
make[2]: Leaving directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples/board_hd77480'
make[2]: Entering directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples/board_simduino'
AVR-CC atmega328p_dummy_blinky.c
    339	     66	      6	    411	    19b	atmega328p_dummy_blinky.axf
../parts/uart_pty.c: In function 'uart_pty_thread':
../parts/uart_pty.c:215:23: warning: ignoring return value of 'write', declared with attribute warn_unused_result [-Wunused-result]
     TRACE(size_t r =) write(p->port[ti].s, buffer, len);
                       ^
../parts/uart_pty.c: In function 'uart_pty_connect':
../parts/uart_pty.c:319:3: warning: ignoring return value of 'system', declared with attribute warn_unused_result [-Wunused-result]
   system(cmd);
   ^
simduino.c: In function 'avr_special_init':
simduino.c:69:2: warning: ignoring return value of 'ftruncate', declared with attribute warn_unused_result [-Wunused-result]
  (void)ftruncate(flash_data->avr_flash_fd, avr->flashend + 1);
  ^
CC simduino.c
CC ../parts/uart_pty.c
LD obj-x86_64-unknown-linux-gnu/simduino.elf
simduino done
rm atmega328p_dummy_blinky.axf
make[2]: Leaving directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples/board_simduino'
make[2]: Entering directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples/board_i2ctest'
AVR-CC atmega1280_i2ctest.c
CC i2ctest.c
CC ../../examples/parts/i2c_eeprom.c
LD obj-x86_64-unknown-linux-gnu/i2ctest.elf
   1147	     16	     12	   1175	    497	atmega1280_i2ctest.axf
i2ctest done
make[2]: Leaving directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples/board_i2ctest'
make[2]: Entering directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples/board_ledramp'
AVR-CC atmega48_ledramp.c
   1841	     32	     11	   1884	    75c	atmega48_ledramp.axf
CC ../parts/button.c
CC ledramp.c
LD obj-x86_64-unknown-linux-gnu/ledramp.elf
ledramp done
make[2]: Leaving directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples/board_ledramp'
make[1]: Leaving directory '/tmp/nix-build-simavr-1.3.drv-0/simavr-51d5fa69f9bc3d62941827faec02f8fbc7e187ab-src/examples'
builder for ‘/nix/store/0yw5ad53c8sdxd7g0j86kadlm7isw5m1-simavr-1.3.drv’ failed with exit code 2
error: build of ‘/nix/store/0yw5ad53c8sdxd7g0j86kadlm7isw5m1-simavr-1.3.drv’ failed
```
